cmd/kubeadm/app/preflight/checks.go

```diff
@@ -514,7 +516,18 @@ func (kubever KubeletVersionCheck) Check() (warnings, errors []error) {
 	if kubeletVersion.LessThan(kubeadmconstants.MinimumKubeletVersion) {
 		return nil, []error{fmt.Errorf("Kubelet version %q is lower than kubadm can support. Please upgrade kubelet", kubeletVersion)}
 	}
-	return nil, []error{}
+
+	if kubever.KubernetesVersion != "" {
+		k8sVersion, err := versionutil.ParseSemantic(kubever.KubernetesVersion)
+		if err != nil {
+			return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+		}
+		firstUnsupportedKubeletVersion := versionutil.MustParseSemantic(fmt.Sprintf("%d.%d.%s", k8sVersion.Major(), k8sVersion.Minor()+1, "0-0"))
+		if kubeletVersion.AtLeast(firstUnsupportedKubeletVersion) {
+			return []error{fmt.Errorf("kubernetes version is lower than kubelet version. This may lead to malfunctional cluster setup. Kubernetes version: %s. Kubelet version: %s", k8sVersion, kubeletVersion)}, nil
```

Comment:
suggest: `The kubelet version is higher than the control plane version. This is not a supported version skew and may lead to a malfunctional cluster. Kubelet version: %s. Control plane version: %s.`

Reply:
Also I think this should be an error

Reply:
I'd suggest to keep it as warning. We don't break every single release interfaces between kubelet and control plane. It might be some cases where it will work incorrectly, but generally it should work. But I can change that to error, not big deal, as soon we will have custom controls on pre-flight checks :)

Reply:
As it never is guaranteed to be a supported state, I want it to be an error. Please fix as above